it/src/main/java/org/corfudb/universe/universe/vm/VmUniverse.java

```diff
@@ -0,0 +1,319 @@
+                // Find the template machine in the inventory
+                VirtualMachine vmTemplate = (VirtualMachine) inventoryNavigator.searchManagedEntity(
+                        ManagedEntityType.VIRTUAL_MACHINE.typeName,
+                        params.getTemplateVMName()
+                );
+
+                log.info("Deploying the VM {} via vSphere {}...", vmName, params.getVSphereUrl());
+
+                // Create customization for cloning process
+                VirtualMachineCloneSpec cloneSpec = createLinuxCustomization(vmName);
+                try {
+                    // Do the cloning - providing the clone specification
+                    Task cloneTask = vmTemplate.cloneVM_Task((Folder) vmTemplate.getParent(), vmName, cloneSpec);
+                    cloneTask.waitForTask();
+                } catch (RemoteException | InterruptedException e) {
+                    throw new UniverseException(String.format("Deploy VM %s failed due to ", vmName), e);
+                }
+                // After the clone task completes, get the VM from the inventory
+                vm = (VirtualMachine) inventoryNavigator.searchManagedEntity(
+                        ManagedEntityType.VIRTUAL_MACHINE.typeName, vmName);
+            }
+            // Ensure we get the VM's IP address before we return its instance
+            String cloneVmIpAddress = null;
+            log.info("Getting IP address for {} from DHCP...please wait...", vmName);
+            while (cloneVmIpAddress == null) {
+                TimeUnit.SECONDS.sleep(5);
+                GuestInfo guest = vm.getGuest();
+                cloneVmIpAddress = guest.getIpAddress();
+            }
+        } catch (RemoteException | MalformedURLException | InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new UniverseException(String.format("Deploy VM %s failed due to ", vmName), e);
+        }
+        return vm;
+    }
+
+    /**
+     * Create the Linux customization for a specific VM to be cloned.
+     *
+     * @return VirtualMachineCloneSpec instance
+     */
+    public VirtualMachineCloneSpec createLinuxCustomization(String cloneName) {
+        VirtualMachineCloneSpec vmCloneSpec = new VirtualMachineCloneSpec();
+
+        //Set location of clone to be the same as template (Datastore)
+        VirtualMachineRelocateSpec vmRelocateSpec = new VirtualMachineRelocateSpec();
+        vmCloneSpec.setLocation(vmRelocateSpec);
+
+        //Clone is powered on, not a template.
+        vmCloneSpec.setPowerOn(true);
+        vmCloneSpec.setTemplate(false);
+
+        //Create customization specs/linux specific options
+        CustomizationSpec customSpec = new CustomizationSpec();
+        CustomizationLinuxOptions linuxOptions = new CustomizationLinuxOptions();
+        customSpec.setOptions(linuxOptions);
+
+        CustomizationLinuxPrep linuxPrep = new CustomizationLinuxPrep();
+        linuxPrep.setDomain("eng.vmware.com");
```

Comment:
These are all very specific constants. Move them to a properties file.

Reply:
fixed